src/briefcase/commands/base.py

```diff
@@ -305,6 +306,20 @@ def validate_locale(self):
 """
             )
 
+    def validate_python_version(self):
+        """Validate the system's python version is not past its EOL date."""
+        python_version = sys.version_info
+        today = date.today()
+        EOL_year = today.year - 2024  # increment this each year past 10-2024
```

Comment:
A short description of the math being used here would be helpful - why is 2024 important? Should this number be updated every year? Also, in any description, it's more helpful to say "October" than "10" because it's clear we're referring to a month.